Cabal/src/Distribution/Backpack/LinkedComponent.hs

```diff
@@ -210,12 +210,19 @@ toLinkedComponent verbosity db this_pid pkg_map ConfiguredComponent {
         -- Read out all the final results by converting back
         -- into a pure representation.
         let convertIncludeU (ComponentInclude dep_aid rns i) = do
-                uid <- convertUnitIdU (ann_id dep_aid)
-                return (ComponentInclude {
-                            ci_ann_id = dep_aid { ann_id = uid },
-                            ci_renaming = rns,
-                            ci_implicit = i
-                        })
+                x <- convertUnitIdU (ann_id dep_aid)
+                case x of
+                    Right uid -> return (ComponentInclude {
+                                    ci_ann_id = dep_aid { ann_id = uid },
+                                    ci_renaming = rns,
+                                    ci_implicit = i
+                                })
+                    Left mod_names ->
+                        let sigsList = hcat $ punctuate (text ", ") $ map (quotes . pretty) mod_names in
+                        failWith $ text "Cannot instantiate requirement in " <+> (pretty $ ann_cname dep_aid)  $$
+                            text "Check that \"build-depends:\" doesn't include library with signatures: " <+> sigsList $$
```

Comment:
I'm not a native speaker, but would "Ensure" instead of "Check" be more precise? Then "that" may be skipped. I'd probably also say "any library". BTW, is it "library" or "package" or "component" or what? How does https://cabal.readthedocs.io/en/latest/ name the things after `build-depends`?

BTW, are the double spaces intentional (the extra one seems to be coming from `<+>`)?